### ISSUE TITLE
[EPIC] Oauth developer applications

### ISSUE THREAD

**Issue by sherifnada** [airbytehq/airbyte#6070]
## Tell us about the problem you're trying to solve
Oauth is an authentication protocol which enables the user (e.g: sherif) of a service (e.g: google) to give a third party service (e.g: Airbyte) access to their data to perform some work on their behalf. 

For example, Sherif can give Airbyte access to his Google Analytics data by using Google's oauth mechanism. 

[This video](https://www.youtube.com/watch?v=KT8ybowdyr0) gives a good overview of why oauth is useful. 

At a high level, the way oauth works is as follows: 

1. A developer obtains a public developer application that allows them to authenticate users
2. The developer implements oauth logic in their service to access user data

This epic is mostly involved with performing the actions needed to obtain public developer applications i.e: step 1 above. 

### Obtaining a public developer application
In order to process user data, a data processor like Airbyte must make itself known to the data custodian e.g: Google. It does this by submitting a request to Google for a **developer application**. This request often involves multiple forms of verification. For example, Google may ask Airbyte to record a video of our application, submit their company name and address (to verify they are a real entity and not hackers/scammers), have a phone call, or perform any other verification. This process involves some back and forth and can often take many days or weeks. This is why it's super important to start the process as early as possible. 

Once the request is approved, Google gives Airbyte some secrets, commonly a **client_id** and **client_secret**, although there may be more or less parameters than just those two. These credentials are tied to a particular developer application, the one that Airbyte applied for. 

A developer application is called a **public developer application** if it can be used to access data on behalf of clients. For example, a Google public developer application owned by Airbyte (the company) can be used in Airbyte Cloud to access users' Google data. On the other hand, a **non-public or private developer application** can typically be used only for testing, either on Airbyte's own data or on a very small set of google users e.g: 100 users. 

In some cases, companies require that a developer like Airbyte create a non-public developer application first, build their integration and test it, then apply separately to make the developer application public (potentially submitting some extra required verification along the way). 

### What this epic is about
This epic is concerned with obtaining public developer applications for all of our connectors. 

There will be multiple steps here: 
For each source connector we support (ignore destinations for now): 
1. Find out if the connector supports oauth
2. if yes, create an issue with the title `Obtain public developer application for <name> connector` and label it with the `requires-oauth` github issue label. 

Once all issues have been created, please add them to your project backlog (GL vetting) and let sherif know to advise on prioritization. Then, in priority order, go through each connector and list the following information: 
1. What is needed to apply for a public developer application that can be used in oauth? This may require you to sign in and play around in the UI. Documentation can be unclear on this topic. 
2. If you can apply for the public developer application, you should do it. You may need to record a demo video of Airbyte or something of the sort. 
3. If at any point you have any questions or need any help, please reach out in one of the shared slack channels

**Please ignore any google, facebook, instagram, amazon, or salesforce connectors. Their oauth applications are already covered.** 

### Some helpful resources
- [company information](https://docs.google.com/document/d/1X3FZFJK9quQGFGFXL1rjF5VOq-EocRF07O2ov2ZL4QU/edit). This will be useful in the verification process. You can only access this via your @airbyte.io email.

**Comment by igrankova** [comment]
Obtain public developer application for amazon ads connector #6129

**Comment by cheyura** [comment]
@sherifnada @igrankova "What is needed to apply for a public developer application that can be used in oauth? This may require you to sign in and play around in the UI." Is this information is needed on every issue? Or if we can create the public dev app promptly it is not necessary.

**Comment by sherifnada** [comment]
@cheyura not needed if it's very simple/obvious

**Comment by igrankova** [comment]
Obtain public developer application for the GitHub source connector #6175
